src/panels/lovelace/cards/hui-weather-forecast-card.ts

```diff
@@ -72,20 +72,61 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
 
   @state() private _config?: WeatherForecastCardConfig;
 
+  @state() private _forecastEvent?: ForecastEvent;
+
+  @state() private _subscribed?: Promise<() => void>;
+
   @property({ type: Boolean, reflect: true, attribute: "veryverynarrow" })
   private _veryVeryNarrow = false;
 
   private _resizeObserver?: ResizeObserver;
 
+  private _handleForecastEvent(forecastEvent: ForecastEvent) {
+    this._forecastEvent = forecastEvent;
+  }
+
+  private _needForecastSubscription() {
+    return (
+      this._config!.forecast_type && this._config!.forecast_type !== "legacy"
```

Comment:
Should this also check `forecast_type !== undefined` ?

Reply:
I don't think that's needed because we already check `this._config!.forecast_type`?